core/src/main/java/com/graphhopper/routing/util/EncodingManager.java

```diff
@@ -613,6 +623,15 @@ public void handleTurnRelationTags(OSMTurnRelation turnRelation, TurnCostParser.
      */
     public IntsRef handleWayTags(ReaderWay way, AcceptWay acceptWay, IntsRef relationFlags) {
         IntsRef edgeFlags = createEdgeFlags();
+        if (customAreaLookup != CustomAreaLookup.EMPTY) {
+            GHPoint estimatedCenter = way.getTag("estimated_center", null);
+            if (estimatedCenter != null) {
+                LookupResult result = customAreaLookup.lookup(estimatedCenter.lat,
+                                estimatedCenter.lon);
+                way.setTag("custom_areas", result.getAreas());
+                way.setTag("spatial_rule_set", result.getRuleSet());
+            }
+        }
```

Comment:
I like the separate handling so that the properties are set before the other encoded value (instead of the `insert=true` code), but can you still move the code back into the specific parser (outside of EncodingManager)?

Reply:
Something like this?

```java
    public IntsRef handleWayTags(ReaderWay way, AcceptWay acceptWay, IntsRef relationFlags) {
        IntsRef edgeFlags = createEdgeFlags();
        CustomAreaParser.handleWayTags(edgeFlags, way, acceptWay.isFerry(), relationFlags, customAreaLookup);
        for (TagParser parser : edgeTagParsers) {
            parser.handleWayTags(edgeFlags, way, acceptWay.isFerry(), relationFlags);
        }
        for (AbstractFlagEncoder encoder : edgeEncoders) {
            encoder.handleWayTags(edgeFlags, way, acceptWay.get(encoder.toString()));
        }
        return edgeFlags;
    }
```